File: project/src/main/java/awale/game/startingMode/Random.java
package awale.game.startingMode;

import awale.game.Bot;
import awale.game.Human;
import awale.game.Player;
import awale.verification.Even;
import awale.verification.Odd;

import java.util.Scanner;

public class Random implements StartingMode {
    @Override
    public void execute(Player[] players, Player p1, Player p2, int[][] board){
        Scanner sc = new Scanner(System.in);
        java.util.Random r = new java.util.Random();
        String name;

        p1.setOpponent(p2);
        p2.setOpponent(p1);
        p1.setBoard(board);
        p2.setBoard(board);

        if (p1 instanceof Human && p2 instanceof Human) {
            System.out.print("player 1 : ");
            name =  sc.nextLine();
            p1.setName(name);
            System.out.print("player 2 : ");
            name =  sc.nextLine();
            p2.setName(name);
        }
        else if (p1 instanceof Human || p2 instanceof Human) {
            System.out.print("bot opponent : ");
            name =  sc.nextLine();
            if(p1 instanceof Human)
                p1.setName(name);
            else
                p2.setName(name);
        }

        if (r.nextInt(2) == 0) {
            p1.setHoleVerify(new Odd());
            p2.setHoleVerify(new Even());
            players[0] = p1;
            players[1] = p2;
        }else {
            p1.setHoleVerify(new Even());
            p2.setHoleVerify(new Odd());
            players[1] = p1;
            players[0] = p2;
        }
    }
}

File: projectC/awale/src/game/startingMode/random.c
#include "../../../headers/game/startingMode/random.h"

void execute_random(Bot bots[2], Bot *b1, Bot *b2, int board[16][3])
{
    // TODO for Human / MQTT

    srand(time(0));

    Verification oddVerifier;
    oddVerifier.satisfy = satisfy_odd;
    Verification evenVerifier;
    evenVerifier.satisfy = satisfy_even;

    if (rand() % 2 == 0)
    {
        setHoleVerify(b1->player, &oddVerifier);
        setHoleVerify(b2->player, &evenVerifier);
        bots[0] = *(b1);
        bots[1] = *(b2);
    }
    else
    {
        setHoleVerify(b1->player, &evenVerifier);
        setHoleVerify(b2->player, &oddVerifier);
        bots[1] = *(b1);
        bots[0] = *(b2);
    }

    setOpponent(b1->player, b2->player);
    setOpponent(b2->player, b1->player);
    // setBoard(b1->player, board);
    // setBoard(b2->player, board);
}
